tests/test_numpy_wavefunction.py

```diff
@@ -69,69 +69,60 @@ def test_var_hermitian(self, tol):
 
         self.assertAlmostEqual(var, expected, delta=tol)
 
-    @pytest.mark.parametrize(
-        "gate", plf.NumpyWavefunctionDevice._operation_map
-    )  # pylint: disable=protected-access
-    def test_apply(self, gate, apply_unitary, tol):
+    def test_apply(self, apply_unitary, tol):
         """Test the application of gates to a state"""
         dev = plf.NumpyWavefunctionDevice(wires=3)
 
-        try:
-            # get the equivalent pennylane operation class
-            op = getattr(qml.ops, gate)
-        except AttributeError:
-            # get the equivalent pennylane-forest operation class
-            op = getattr(plf, gate)
-
-        # the list of wires to apply the operation to
-        w = list(range(op.num_wires))
-
         obs = qml.expval(qml.PauliZ(0))
-        if op.par_domain == "A":
-            # the parameter is an array
-            if gate == "QubitUnitary":
-                p = np.array(U)
-                w = [0]
+
+        ops = [
```

Comment:
The previous version seems to have been using parametrization. It would be good to keep that style because that way we can be sure which gate application fails if the test fails.